### PR TITLE
28-crud

### PR DESCRIPTION
**Pull request by wojciechkapko**
Added Create, Update, Remove options

Single view (Create) where we handle both creating and editing
Remove action does not have a view, we could add some kind of a popup with confirmation later on

### CODE REVIEW

File: DrinkManagerWeb/Controllers/DrinksController.cs [hunk 1]
```diff
@@ -50,5 +52,132 @@ public IActionResult DrinkDetails(string id)
...
+            if (drink == null)
+            {
+                return NotFound();
+            }
+            _db.Drinks.Remove(drink);
+
+            TempData["Alert"] = $"Drink {drink.Name} removed";
+
+            return RedirectToAction(nameof(Index));
+        }
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Controllers/DrinksController.cs:207]: I think here should be question for user: "Are you sure you want to remove drink.Name?

File: DrinkManagerWeb/Views/Drinks/Create.cshtml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    <input type="text" class="form-control" asp-for="Category" placeholder="Category">
+                    <span asp-validation-for="Category" class="text-danger"></span>
+                </div>
+            </div>
+            <div class="row">
+                <div class="form-group col-md-6">
+                    <label asp-for="AlcoholicInfo">Alcohol content</label>
+                    <select class="form-control" asp-for="AlcoholicInfo">
+                        <option>Alcoholic</option>
+                        <option>No alcohol</option>
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Views/Drinks/Create.cshtml]: Here is a typo, should be "Non alcoholic"

File: DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        [StringLength(25)]
+        public string Name { get; set; }
+        [Required]
+        [StringLength(25)]
+        public string Category { get; set; }
+        public string AlcoholicInfo { get; set; }
+        [StringLength(25)]
+        public string GlassType { get; set; }
+        [Required]
+        public string Instructions { get; set; }
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs:23]: When you don't type any instructions, information that you should type someting appears before text "Image URL" and moved it.

File: DrinkManagerWeb/Views/Drinks/Create.cshtml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                        <div class="form-group mb-3">
+                            <input type="text" class="form-control" asp-for="Ingredient1" placeholder="Ingredient">
+                            <span asp-validation-for="Ingredient1" class="text-danger"></span>
+                        </div>
+                    </div>
+                    <button type="button" onclick="AddInputAndRestartValidation()" class="btn btn-primary"><i class="fas fa-plus"></i> Add Ingredient</button>
+                </div>
+            </div>
+            <div class="mt-3 border-top">
+                <input class="btn btn-primary mt-3" type="submit" value="Save" />
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Views/Drinks/Create.cshtml:71]: When drink has more than 7 ingredients this button is hide under footer.

File: DrinkManagerWeb/Views/Drinks/Create.cshtml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    <script>
+        function AddIngredientInput(value = "") {
+            let ingredientNumber = $("#ingredients-group input").length + 1;
+            if (ingredientNumber > 15) {
+                if ($(".alert").length === 0) {
+                    $("form").prepend(
+                        `<div class="alert alert-danger alert-dismissible" role="alert">
+                                        <button type="button" class="close" data-dismiss="alert">&times;</button>
+                                        You can add maximum of 15 ingredients
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Views/Drinks/Create.cshtml]: Missing dot in the end of this sentence. Is it possible for this information to appear under the button "Add Ingredient"?

File: DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+        [Required]
+        [StringLength(25)]
+        public string Name { get; set; }
+        [Required]
+        [StringLength(25)]
+        public string Category { get; set; }
+        public string AlcoholicInfo { get; set; }
+        [StringLength(25)]
+        public string GlassType { get; set; }
```
Comment (joanna-bukowska) [inline DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs:21]: GlassType is not required. I think it should be.

File: DrinkManagerWeb/Views/Drinks/Create.cshtml (hunk 2 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Views/Drinks/Create.cshtml]: fixed

File: DrinkManagerWeb/Controllers/DrinksController.cs (hunk 1 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Controllers/DrinksController.cs:207]: Modal with such question has been added

File: DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs (hunk 3 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs:23]: fixed

File: DrinkManagerWeb/Views/Drinks/Create.cshtml (hunk 4 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Views/Drinks/Create.cshtml:71]: footer has been removed 🤯

File: DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs (hunk 6 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Models/ViewModels/DrinkCreateViewModel.cs:21]: done

File: DrinkManagerWeb/Views/Drinks/Create.cshtml (hunk 5 above)
Comment (wojciechkapko) [reply DrinkManagerWeb/Views/Drinks/Create.cshtml]: done

**Review by marcin-sieminski — APPROVED** [review]

**Review by joanna-bukowska — APPROVED** [review]

**Review by joanna-bukowska — APPROVED** [review]

**Pull request merged**